sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -323,6 +326,59 @@ export class BatchingReceiverLite {
     }
   }
 
+  private async drainReceiverIfNeeded(
+    receiver: MinimalReceiver,
+    loggingPrefix: string,
+    abortSignal?: AbortSignalLike
+  ): Promise<void> {
+    // Drain any pending credits.
+    if (receiver.isOpen() && receiver.credit > 0) {
+      let drainTimedout: boolean = false;
+      let drainTimer: ReturnType<typeof setTimeout>;
+      const drainPromise = new Promise<void>((resolve) => {
```

Comment:
using `createAbortablePromise` and utilizing the `cleanupBeforeAbort` option should help hide the handling of the abort signal.

Reply:
I had looked at `createAbortablePromise` too but didn't use it because it throws and I would have to deal handle the error. It may be nice if I could override `onAbort` to just resolve.

Reply:
Will re-visit this after the discussion in issue #25290